### PR TITLE
fix: close hamburger menu when click on item on mobile

### PR DESCRIPTION
**Pull request by coratgerl**
## Summary

When we click on an item in an hamburger menu, we should close the menu after click event.

## Related Issue

Fixes: https://github.com/web-infra-dev/rspress/issues/1711

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: packages/theme-default/src/components/NavHambmger/index.tsx [hunk 1]
```diff
@@ -18,13 +18,14 @@ export function NavHamburger(props: Props) {
...
         isScreenOpen={isScreenOpen}
+        toggleScreen={toggleScreen}
         siteData={siteData}
         pathname={pathname}
       />
       <button
         onClick={toggleScreen}
         aria-label="mobile hamburger"
-        className={`${isScreenOpen ? styles.active : ''} ${
+        className={`${isScreenOpen ? styles.active : ''} buttonHamburger ${
```
Comment (Timeless0911) [inline packages/theme-default/src/components/NavHambmger/index.tsx:28]: `rspress-mobile-hamburger` is better?

File: packages/theme-default/src/components/NavScreen/index.tsx [hunk 2]
```diff
@@ -103,7 +105,7 @@ export function NavScreen(props: Props) {
   }, [isScreenOpen]);
   return (
     <div
-      className={`${styles.navScreen} ${isScreenOpen ? styles.active : ''}`}
+      className={`${styles.navScreen} ${isScreenOpen ? styles.active : ''} navScreen`}
```
Comment (Timeless0911) [inline packages/theme-default/src/components/NavScreen/index.tsx:108]: `rspress-nav-screen`

**Comment by SoonIter** [comment]
LGTM after adopting the suggestions

File: packages/theme-default/src/components/NavHambmger/index.tsx (hunk 1 above)
Comment (coratgerl) [reply packages/theme-default/src/components/NavHambmger/index.tsx:28]: Yes you right

File: packages/theme-default/src/components/NavScreen/index.tsx (hunk 2 above)
Comment (coratgerl) [reply packages/theme-default/src/components/NavScreen/index.tsx:108]: Yes you right

**Comment by coratgerl** [comment]
Done ! Thanks for your feedbacks !

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**